projects/laji/src/app/+taxonomy/taxon/info-card/taxon-identification/identification-list/identification-list.component.ts

```diff
@@ -87,24 +87,50 @@ export class IdentificationListComponent implements OnDestroy {
     this.destroyMouseupListener();
   }
 
-  openImage(index) {
+  openImage(index?: number) {
     this.overlayLoader
       .attach(ImageModalOverlayComponent)
       .to('body')
       .show({isAnimated: false});
     this.showOverlay = true;
     this.overlayRef = this.overlayLoader._componentRef;
-    const [filteredIndex, filteredChildren] = indexAndArrAfterFilter(
-      index, this.taxon.children,
-      taxonomy => taxonomy.multimedia && taxonomy.multimedia.length > 0
-    );
-    this.overlayRef.instance.modalImages = filteredChildren.map(taxonomy => <Image>{
-        ...taxonomy.multimedia[0],
-        taxonId: taxonomy.id,
-        vernacularName: taxonomy.vernacularName,
-        scientificName: taxonomy.scientificName
+
+    if (this?.taxon?.species) {
+      const modalImages = [];
+      this.taxon.multimedia.forEach((media, i) => {
+        modalImages.push(<Image>{
+          ...this.taxon.multimedia[i],
+          taxonId: this.taxon.id,
+          vernacularName: this.taxon.vernacularName,
+          scientificName: this.taxon.scientificName
+        });
       });
-    this.overlayRef.instance.showImage(filteredIndex);
+      this.overlayRef.instance.modalImages = modalImages;
```

Comment:
Not very important, but this mapping could be done without the modalImages array by using `Array.map`

Reply:
Should be fixed now!